mina.netty/src/main/java/org/jboss/netty/channel/socket/nio/NioChildDatagramPipelineSink.java

```diff
@@ -71,25 +71,18 @@ public void eventSunk(final ChannelPipeline pipeline, final ChannelEvent e)
                     break;
             }
         } else if (e instanceof MessageEvent) {
-            // Making sure that child channel WriteFuture is fired on child channel's worker thread
             final MessageEvent childMessageEvent = (MessageEvent) e;
             ParentMessageEvent parentMessageEvent = new ParentMessageEvent(childMessageEvent);
-            ChannelFuture parentFuture = parentMessageEvent.getFuture();
-            parentFuture.addListener(f -> {
-                childChannel.getWorker().executeInIoThread(() -> {
-                    if (f.isSuccess()) {
-                        childFuture.setSuccess();
-                    } else {
-                        childFuture.setFailure(f.getCause());
-                    }
-                });
-            });
 
             // Write to parent channel
             NioDatagramChannel parentChannel = (NioDatagramChannel) childChannel.getParent();
             boolean offered = parentChannel.writeBufferQueue.offer(parentMessageEvent);
             assert offered;
             parentChannel.worker.writeFromUserCode(parentChannel);
+
+            // No need to propagate parentMessageEvent.getFuture() to childFuture
+            // as UDP is unreliable.
+            childFuture.setSuccess();
```

Comment:
From John's comment I think you may need to schedule this onto the child io thread, something like 
```
childChannel.getWorker().executeInIoThread(() -> {
                        childFuture.setSuccess();
});
```